File: pat1007.py
def generateP( num , n ):
	for x in range(num+1, 2*n):
		if ( isP(x)):
			return x
			break

def isP( m ):
	n = pingfang(m)
	if( m == 2 or m == 3):
		return 1
	else :
		#y = 0
		for x in range(2, n+1 ):
			if ( (m % x) == 0):
				return 0
			'''else :
				y = x'''
		if ( x == n ):
			return 1

def pingfang( shuzi ):
	x= float(shuzi)
	y= sqrt(x)
	return int(y)
	pass


import time
from math import sqrt
number = int( input() )
starttime = time.clock()
count = 0 
p1 = 2
p2 = 3
while ( p2 <= number ):
	if ( (p2 - p1) == 2):
		count += 1
	p1 = p2
	p2 = generateP(p2 , number)
else :
	print(count)
endtime = time.clock()
print(endtime - starttime)

File: pat1007.c
#include <stdio.h>
#include <math.h>

int main(void)
{
	int number = 0;
	int count = 0;
	int p1 = 2;
	int p2 = 3;
	int m = 0;
	scanf("%d", &number);
	while( p2 <= number){
		if (p2 - p1 == 2)
		{
			count++;
		}
		p1 = p2;
		p2 = generateP(p2, number);
	}
	printf("%d", count);
	return 0;
}

int generateP ( int num , int n){
    int i = 0;
	for (i = num+1; i < 2*n; i++)
	{
		if (isP(i))
		{
			return i;
		}
	}
}

int isP( int m){
    int n = 0;
    n = sqrt (m);
    int i = 0;
	if (m == 2 || m == 3)
	{
		return 1;
	}
	else
	{
		for (i = 2; i <= n; i++)
		{
			if (m % i == 0)
			{
				return 0;
			}
		}
		if (i == n+1 )
		{
			return 1;
		}
	}
}
